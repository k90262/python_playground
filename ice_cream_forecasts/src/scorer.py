import enum
import random


class IceCream(enum.Enum):
    Strawberry = 1
    Chocolate = 2
    Vanilla = 3


flavour = None

def get_score():
    sunny_today = lookup_weather()
    if flavour == IceCream.Strawberry:
        if sunny_today:
            return 10
        else:
            return 5
    elif flavour == IceCream.Chocolate:
        return 6
    elif flavour == IceCream.Vanilla:
        if sunny_today:
            return 7
        else:
            return 5
    else:
        return -1


def lookup_weather():
    # placeholder implementation - real version would make API call to weather service
    sunny = random.choice([True, False])
    return bool(sunny)


def update_selection():
    score = get_score()
    if score > 5:
        global flavour
        # placeholder implementation - real version would use machine learning to predict sales
        flavour = random.choice([IceCream.Strawberry, IceCream.Chocolate, IceCream.Vanilla])


def get_sales_forecast():
    forecasts = {
        IceCream.Strawberry: 9,
        IceCream.Vanilla: 11,
        IceCream.Chocolate: 3
    }
    return forecasts[flavour]
