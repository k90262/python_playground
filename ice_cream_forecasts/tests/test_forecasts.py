import datetime

import approvaltests

from forecasts import print_sales_forecasts_with_update_selection
from scorer import IceCream
import scorer


def test_sales_forecasts(capsys):
    print_sales_forecasts_with_update_selection(
        now=datetime.datetime.fromisoformat("2025-01-08 18:08:53.024945"),
        update_selection=stub_update_selection
    )
    output = capsys.readouterr()
    approvaltests.verify(output.out)

def stub_update_selection():
    scorer.flavour = IceCream.Vanilla