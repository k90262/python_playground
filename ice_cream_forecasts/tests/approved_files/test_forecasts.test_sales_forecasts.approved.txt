Forecast at time 2025-01-08 18:08:53.024945
Steve score: 9
Julie score: 11
Francis score: 11
